I1463 open shift connector - ready for the review (#1080)
codenvy/che-docs: src/main/pages/che-7/extensions/proc_creating-components-with-openshift-connector-in-eclipse-che.adoc

---
title: Creating Components with OpenShift Connector in Eclipse Che
keywords:
tags: []
sidebar: che_7_docs
permalink: che-7/creating-components-with-openshift-connector-in-eclipse-che/
folder: che-7/extensions
summary:
---
// using-openshift-connector-in-eclipse-che

[id="creating-components-with-openshift-connector-in-eclipse-che_{context}"]

> 

Reviewer: ```suggestion
```

= Creating Components with OpenShift Connector in {prod-short}

> In the context of OpenShift, Components and Services, are basic structures that need to be stored in Application, which is a part of the OpenShift Project that organizes deployables into virtual folders for better readability.

Reviewer: ```suggestion
In the context of OpenShift, Components and Services are basic structures that need to be stored in Application, which is a part of the OpenShift project that organizes deployables into virtual folders for better readability.
```

> This chapter describes how to create OpenShift Components in the {prod-shor} using the OpenShift Connector plug-in and push them in to the OpenShift cluster.

Reviewer: ```suggestion
This chapter describes how to create OpenShift Components in the {prod-shor} using the OpenShift Connector plug-in and push them to an OpenShift cluster.
```

.Prerequisites

* A running instance of {prod-short}. To install an instance of {prod-short}, see link:{site-baseurl}che-7/che-quick-starts/[{prod-short} quick-starts].

> * User is logged in to the OpenShift cluster using the Openshift Connector plug-in.

Reviewer: ```suggestion
* The user is logged in to an OpenShift cluster using the Openshift Connector plug-in.
```

.Procedure

> . In the Openshift Connector panel, right-click the row with the red OpenShift icon, and select *New Project*.

Reviewer: ```suggestion
. In the OpenShift Connector panel, right-click the row with the red OpenShift icon and select *New Project*.
```

. Enter a name for your project.
. Right-click the created project and select *New Component*.
+

> You will be prompted to create a new application, in which the component can be stored.

Reviewer: ```suggestion
```

Author: Why to remove this line? Its actually like that. If you want to create a Component, you will need to create an Application first. (it's a little bit counterintuitive, but it is like it is). Approved by OpenShift Connector QE team lead.

I will fix those uppercases for those OpenShit components, though.

> . Enter the name for the application.

Reviewer: ```suggestion
. When prompted, enter the name for a new application in which the component can be stored.
```

+
The following options of source for your component are displayed:
+
.. *Git Repository*
+

> This prompts you to fill in a git repository URL and select the intended revision of the runtime.

Reviewer: ```suggestion
This prompts you to fill in a Git repository URL and select the intended revision of the runtime.
```

.. *Binary File*
+
This prompts you to select a file from the file explorer.
.. *Workspace Directory*
+
This prompts you to select a folder from the file explorer.

> . Enter the name for your component.

Reviewer: ```suggestion
. Enter the name for the component.
```

. Select the component type.
+
// this is the runtime - which is a middleware used for running a source code
. Select the component type version.
+
//runtime version
. The component is created. Right-click the component, select *New URL*, and enter a name of your choice.
+
//creates a route with the given name in openshift

> . The component is ready to be pushed into the OpenShift cluster. To do so, right-click the component and select *Push*.

Reviewer: ```suggestion
. The component is ready to be pushed to the OpenShift cluster. To do so, right-click the component and select *Push*.
```

+

> The component is now deployed to the cluster. You may right-click for selecting additional actions like debug, open in a browser. However, this requires port 8080 to be exposed.

Reviewer: ```suggestion
The component is now deployed to the cluster. Right-click for selecting additional actions, such as debugging and opening in a browser (requires port `8080` to be exposed).
```
